smart-frontend/app/src/store/authSlice.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createSlice } from '@reduxjs/toolkit';
+import type { RequestError } from '@api';
+import { AuthApi } from '@api';
+import { createAsyncThunk } from './redux';
+import { getErrorMessage } from '@utils/responseUtils';
+import { showError } from './notificationsSlice';
+
+type LoginActionPayload = {
+  username: string;
+  password: string;
+};
+
+export enum AUTH_STATE {
+  Checking = 'checking',
+  NotAuth = 'not_auth',
+  Authed = 'authed',
+}
+export type AuthState = AUTH_STATE.NotAuth | AUTH_STATE.Checking | AUTH_STATE.Authed;
+
+type UserState = {
+  username: string;
+  needCheckSession: boolean;
+  hasError: boolean;
+  authState: AuthState;
```

Comment:
can we use AUTH_STATE here? Or as an option we can try to delete AUTH_STATE enum and use
type AuthState = "NotAuth" | "Checking" | "Authed"

Reply:
good idea, thanks!